addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -886,6 +887,155 @@ QUnit.test('current partner notify is typing again to other members every 50s of
+        "attachment displayed should be uploaded"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Composer_buttonSend',
+        "composer send button should still be present"
+    );
+    assert.ok(
+        !document.querySelector('.o_Composer_buttonSend').attributes.disabled,
+        "composer send button should be enabled as attachment is now uploaded"
+    );
+});
+
+QUnit.test('warning when attempting to post message when attachments are still uploading', async function (assert) {
+    assert.expect(8);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const res = this._super(...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await attachmentUploadedPromise;
+            }
+            return res;
+        },
+        services: {
+            notification: {
+                notify(params) {
+                    assert.strictEqual(
+                        params.message,
+                        "Please wait while the file is uploading.",
+                        "notification content should be about the uploading file"
+                    );
+                    assert.strictEqual(
+                        params.title,
+                        "Uploading error",
+                        "notification title should be 'uploading error'"
+                    );
+                    assert.strictEqual(
+                        params.type,
+                        'warning',
+                        "notification should be a warning"
+                    );
+                    assert.step('notification');
+                }
+            }
+        },
+    });
+    const thread = this.env.models['mail.thread'].create({
+        composer: [['create', { isLog: false }]],
+        id: 20,
+        model: 'res.partner',
+    });
+    await this.createComposerComponent(thread.composer);
+    const file = await createFile({
+        content: 'hello, world',
+        contentType: 'text/plain',
+        name: 'text.txt',
+    });
+    await afterNextRender(() =>
+        inputFiles(
+            document.querySelector('.o_FileUploader_input'),
+            [file]
+        )
+    );
+
+    assert.containsOnce(
+        document.body,
+        '.o_Attachment',
+        "should have only one attachment"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Attachment.o-temporary',
+        "attachment displayed is being uploaded"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Composer_buttonSend',
+        "composer send button should be displayed"
+    );
+
+    // Try to send message
+    document
+        .querySelector(`.o_ComposerTextInput_textarea`)
+        .dispatchEvent(new window.KeyboardEvent('keydown', { key: 'Enter' }))
+    ;
+    assert.verifySteps(
+        ['notification'],
+        "a notification should have been fired"
```

Comment:
I would be more specific here. Imagine test fails due to not passing to Notification Service. This doesn't explain why it's important that we have to pass to this step.

Reply:
I don't understand what you want to add ? The flow is expected to generate a notification, if it doesn't then the test fails and  you need to look at it, what is missing ?

Reply:
Would clarify in assertion comment that it should display a notification to tell message cannot be posted when attachments are still being uploaded.

Yes, assertions that are intercepted state that, but if it fails to being caught then it's hard to know why this failed assertion is important on the QUnit interface.

Sure, you could argue we just have to read the code of the test... But it's not nice to at `.displayEvent('keydown, '{ key: Enter })` and understand that we must look at the top of test in the intercept... This is an unfortunate necessity due to notification being hardly testable due to being a service.

Ideally, it should display in DOM and the assertion should look in the DOM, and in that case we wouldn't have a need for `.step()`/`.verifySteps()` and have to consider the case that `.step()` is not registered due to potential bugs in connecting stuffs together (services/components/widgets/etc.).

Reply:
My 1st comment was badly formulated, my bad ^^'

Reply:
No problemo, I got the idea now ;p